nuxeo-runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/BackingServiceConfigurator.java

```diff
@@ -0,0 +1,265 @@
+                    }
+
+                } catch (IOException e) {
+                    log.warn("Unable to read check configuration for template : " + template, e);
+                } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+                    throw new ConfigurationException("Unable to check configuration for backing service " + template,
+                            e);
+                }
+            }
+            checkers.add(new DBCheck());
+        }
+        return checkers;
+    }
+
+    /**
+     * Read the classpath parameter from the template and expand parameters with their value. It allow classpath of the
+     * form ${nuxeo.home}/nxserver/bundles/... VisibleForTesting
+     *
+     * @param template The name of the template
+     * @return
+     */
+    String getClasspathForTemplate(String template) {
+        String classPath = configurationGenerator.getUserConfig().getProperty(template + ".check.classpath");
+        final TextTemplate templateParser = new TextTemplate(configurationGenerator.getUserConfig());
+        classPath = templateParser.processText(classPath);
+        return classPath;
+    }
+
+    /**
+     * Build a ClassLoader based on the classpath definition of a template.
+     *
+     * @param templateDir
+     * @param classPath
+     * @return
+     * @throws ConfigurationException
+     * @throws IOException
+     * @since 9.2
+     */
+    protected Optional<URLClassLoader> getClassLoaderForTemplate(File templateDir, String classPath)
+            throws ConfigurationException, IOException {
+        if (StringUtils.isBlank(classPath)) {
+            return Optional.empty();
+        }
+
+        String[] classpathEntries = classPath.split(":");
+
+        List<URL> urlsList = new ArrayList<>();
+
+        List<File> files = new ArrayList<>();
+        for (String entry : classpathEntries) {
+            files.addAll(getJarsFromClasspathEntry(templateDir, entry));
+        }
+
+        if (!files.isEmpty()) {
+            for (File file : files) {
+                try {
+                    urlsList.add(new URL("jar:file:" + file.getPath() + "!/"));
+                    log.debug("Added " + file.getPath());
+                } catch (MalformedURLException e) {
+                    log.error(e);
+                }
+            }
+        } else {
+            return Optional.empty();
+        }
+
+        URLClassLoader ucl = new URLClassLoader(urlsList.toArray(new URL[0]));
+        return Optional.of(ucl);
+    }
+
+    /**
+     * Given a single classpath entry, return the liste of JARs referenced by it.<br>
+     * For instance :
+     * <ul>
+     * <li>nxserver/lib -> ${templatePath}/nxserver/lib</li>
+     * <li>/somePath/someLib-*.jar</li></li> VisibleForTesting
+     *
+     * @param templatePath
+     * @param entry
+     * @return
+     */
+    Collection<File> getJarsFromClasspathEntry(File templatePath, String entry) {
```

Comment:
This method (and everywhere else you can) should not use `File` but `Path`, which is much more versatile. In particular you don't need to define your own `isInSubDirectory` because you can use `Path.startsWith` for comparisons.